src/Features/Core/Portable/SolutionCrawler/IdleProcessor.cs

```diff
@@ -86,11 +124,13 @@ private async Task ProcessAsync()
                 {
                     // wait for next item available
                     await WaitAsync(CancellationToken).ConfigureAwait(false);
+                    CancellationToken.ThrowIfCancellationRequested();
 
                     using (Listener.BeginAsyncOperation("ProcessAsync"))
                     {
                         // we have items but workspace is busy. wait for idle.
                         await WaitForIdleAsync(Listener).ConfigureAwait(false);
+                        CancellationToken.ThrowIfCancellationRequested();
```

Comment:
added early exits so that we don't even try to start work if we're shutting down.